lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -399,6 +399,12 @@ def _get_subnet_ids_from_subnet_list(subnet_list):
     return subnet_id_list
 
 
+def get_valid_listener_parameters(listener):
+    # See CreateListenerInput in https://github.com/boto/botocore/blob/f51271ac3ce4947ab3bbe46b82bdd79d8be0d5c2/botocore/data/elbv2/2015-12-01/service-2.json
+    valid_parameters = ['LoadBalancerArn', 'Protocol', 'Port', 'DefaultActions', 'SslPolicy', 'Certificates']
+    return {key: value for key, value in listener.items() if key in valid_parameters}
```

Comment:
Just a driveby comment: for pyhton-2.6 compatibility we cannot use dict comprehensions.  However, this syntax is almost equivalent:
```
return dict((key, value) for key, value in listener.items() if key in valid_parameters)
```